docs/llvm/pass/index.md

```diff
@@ -0,0 +1,9 @@
+# Pass
+
+The LLVM pass framework is an important part of the LLVM system, because LLVM passes are where the most interesting parts of the compiler exist. Passes perform transformations and optimizations that make up the compiler, they build the analysis results that are used by these transformations, and they are, above all, a structuring technique for compiler code.
+
+An LLVM pass can be regarded as a black box that takes LLVM IR as input and produces outputs depending on what kind the pass is. There're two kinds of passes: **transformation pass** and **analysis pass**. For a transformation pass, it modifies the input IR and produces the transformed IR. For an analysis pass, it does not modify the input IR, but it looks into the IR and produces certain information that describes some properties about the input IR. Transformation passes typically implement compiler optimizations, program transformation, program instrumentation and other tasks. Analysis passes typically provide necessary information for the transformation passes to work.
```

Comment:
> For a transformation pass, it modifies the input IR and produces the transformed IR.

What about backend passes? (I mean, they may transform the IR to DAGs/MachineInstr/MCInstr etc.)